Login setup w session token

HelenaLykstoft opened the pull request:
Resolves #5 
I have built an authentication flow, with a TokenValidationMiddleware that validates the token for the requests ( and adds the UserId to the HTTPContext if the token is valid )
SessionStore manages the sessions, and stores the token with the user and expiry time.
I changed swagger to have a bearer token, so right now you need to insert that specific token you get from login, so swagger always "has" it ( or as soon as it is valid and havent expired ) 
- When you run the http page, there is a authentication button. After using login endpoint, you take the token and insert it there. Im not sure there is a better way of doing this, we could put it in a db but that is just unsafe i would say.

HelenaLykstoft committed:
Login setup w session token

HelenaLykstoft committed:
Spelling error in workflow file

HelenaLykstoft committed:
Test workflow

HelenaLykstoft committed:
Try fix for token, not working yet.

HelenaLykstoft committed:
Login, logout and me endpoint now working

Resolves #5
- I used swaggers own authorization header, dont know if this can be sent along with kafka.
- Currently working on adding tests to login, logout and the me endpoint

HelenaLykstoft committed:
Updated tests

Tests now works

HelenaLykstoft committed:
Cleanup, tried other tests but didnt work

HelenaLykstoft committed:
Update TokenValidationMiddleware

- Ignores header in create now too

CarolineHoeg reviewed (changes requested):
Har opdaget nogle /create problemer, og muligvis er der et problem med login igen og igen. Har sendt dokumentation af fejl i privat forum grundet hemmelige credentials.

HelenaLykstoft committed:
Review changes

- Add Not empty for all validation rules for register user
- Minor changes to AccountService, so there wont be added to the database before every rule etc is fulfilled
- Adds a ReadOnly for the UserTypeId, so you cannot change it. ( It is always just 1 ) ( Adding ability to change if you are admin, later )

HelenaLykstoft committed:
Refactor changes

- Started working on making a check to make sure that if you are logged in and have a token, that you dont get a new token. Had to delete it all because it became a bigger issue than anticipated so starting over

HelenaLykstoft committed:
Changed tokens per user

- A user can now only have one token.

HelenaLykstoft committed:
Commit for tests to run again

HelenaLykstoft committed:
Test changes

HelenaLykstoft committed:
Test changes 2

HelenaLykstoft committed:
Test change again

HelenaLykstoft committed:
Changed the createuser test

CarolineHoeg reviewed (approved):
Jeg synes at det ser fint ud 😄
